corehq/pillows/app_submission_tracker.py

```diff
@@ -126,3 +132,90 @@ def get_sql_app_form_submission_tracker_reindexer():
         FormReindexAccessor(include_attachments=False)
     )
     return AppFormSubmissionReindexer(doc_provider, FORM_SQL, 'form_processor_xforminstancesql')
+
+
+@quickcache(['domain'], timeout=60 * 60 * 5)
+def _get_apps_for_domain(domain):
+    # thin cached wrapper for get_app_ids_in_domain
+    return get_app_ids_in_domain(domain)
+
+
+class UserAppFormSubmissionDocProcessor(BaseDocProcessor):
+    def __init__(self, pillow_processor):
+        self.pillow_processor = pillow_processor
+
+    def process_doc(self, doc):
+        form_submission_changes = self._doc_to_changes(doc)
+        for change in form_submission_changes:
+            try:
+                self.pillow_processor.process_change(None, change)
+            except Exception:
+                return False
+        return True
+
+    def handle_skip(self, doc):
+        print('Unable to process user {}'.format(
+            doc['_id'],
+        ))
+        return True
+
+    def _doc_to_changes(self, doc):
+        # creates a change object for the last form submission
+        # for the user to each of their apps.
+        # this allows us to reindex for the app status report
+        # without reindexing all forms.
+        changes = []
+        apps = []
+        if doc['doc_type'] == 'CommCareUser':
+            apps = _get_apps_for_domain(doc['domain'])
+        else:
+            for domain in (dm['domain'] for dm in doc['domain_memberships']):
+                apps = _get_apps_for_domain(domain)
```

Comment:
This should presumably be appending to the list and not overwriting it each time